pkg/kubelet/prober/prober_manager.go

```diff
@@ -298,24 +304,31 @@ func (m *manager) UpdatePodStatus(pod *v1.Pod, podStatus *v1.PodStatus) {
 		var ready bool
 		if c.State.Running == nil {
 			ready = false
-		} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok && result == results.Success {
-			ready = true
+		} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
+			ready = result == results.Success
+			if !ready {
+				m.tryTriggerWorker(pod.UID, c.Name, readiness)
+			}
 		} else {
 			// The check whether there is a probe which hasn't run yet.
-			w, exists := m.getWorker(pod.UID, c.Name, readiness)
-			ready = !exists // no readinessProbe -> always ready
-			if exists {
-				// Trigger an immediate run of the readinessProbe to update ready state
-				select {
-				case w.manualTriggerCh <- struct{}{}:
-				default: // Non-blocking.
-					klog.InfoS("Failed to trigger a manual run", "probe", w.probeType.String())
-				}
```

Comment:
and call that function only where its result is used, e.g. here

Reply:
we can  wrap it into function, but call it here is not a good idea, because it's in the for loop here.